framework/src/controller/application.js

```diff
@@ -255,6 +259,20 @@ class Application {
 		__private.transactions.set(this, transactions);
 	}
 
+	/**
+	 * Register migrations with the application
+	 *
+	 * @param {Object} namespace - Migration namespace
+	 * @param {Array} migrations - Migrations list. Format ['yyyyMMddHHmmss_name_of_migration.sql']
+	 */
+	registerMigrations(namespace, migrations) {
+		assert(namespace, 'Namespace is required');
+		assert(migrations instanceof Array, 'Migrations list should be an array');
+		const currentMigrations = this.getMigrations() || {};
+		currentMigrations[namespace] = migrations;
```

Comment:
I don't like this approach as it allows the app to register module migrations outside the `registerModule` method

Reply:
Yeah one can register it any case because `registerMigrations` is on the app level. And we are not limiting that call to track either it was called from registerModule or outside.

Reply:
Before registering the migration we are checking if the `namespace` was already used. That way you can't register module migrations twice (which is good imo)

Reply:
That's a valid case for modules, but you can register migrations from the app level as well. So one can call `app.registerMigrations` as many times he want. And with each call we override his old register migrations is not a good approach. And we can't restrict developer to call `app.registerMigrations` only once.

Reply:
> Before registering the migration we are checking if the `namespace` was already used. That way you can't register module migrations twice (which is good imo)

Actually I've just realised I'm not checking if the `namespace` was registered before. I'll add this validation and it should be good to go imo.

Reply:
I don't think a solid usecase in running registerMigrations twice, and if the computation is required, it should be able to be done prior to calling the first one.
We can implement this feature when we need them